src/benchmark/benchmark_map.cpp

```diff
@@ -0,0 +1,177 @@
+// Luanti
+// SPDX-License-Identifier: LGPL-2.1-or-later
+// Copyright (C) 2023 Minetest Authors
+
+#include "catch.h"
+#include "dummygamedef.h"
+#include "map.h"
+#include "mapsector.h"
+
+class TestMap : public Map {
+public:
+	TestMap(IGameDef *gamedef) : Map(gamedef) {}
+
+	MapBlock * createBlockTest(v3s16 p)
+	{
+		v2s16 p2d(p.X, p.Z);
+		s16 block_y = p.Y;
+
+		MapSector *sector = getSectorNoGenerate(p2d);
+		if (!sector) {
+			sector = new MapSector(this, p2d, m_gamedef);
+			m_sectors[p2d] = sector;
+		}
+
+		MapBlock *block = sector->getBlockNoCreateNoEx(block_y);
+		if (block)
+			return block;
+
+		return sector->createBlankBlock(block_y);
+	}
+
+};
+
+static void fillMap(TestMap &map, s16 n)
+{
+	for(s16 z=0; z<n; z++)
+	for(s16 y=0; y<n; y++)
+	for(s16 x=0; x<n; x++) {
+		v3s16 p(x,y,z);
+		// create an empty block
+		map.createBlockTest(p);
+	}
+}
+
+static int readBlocks(Map &map, s16 n)
+{
+	int result = 0;
+	for(s16 z=0; z<n; z++)
+	for(s16 y=0; y<n; y++)
+	for(s16 x=0; x<n; x++) {
+		v3s16 p(x,y,z);
+		MapBlock *block = map.getBlockNoCreateNoEx(p);
+		if (block) {
+			result++;
+		}
+	}
+	return result;
+}
+
+static int readRandomBlocks(Map &map, s16 n)
+{
+	int result = 0;
+	for(int i=0; i < n * n * n; i++) {
+		v3s16 p(myrand_range(0, n), myrand_range(0, n), myrand_range(0, n));
+		MapBlock *block = map.getBlockNoCreateNoEx(p);
+		if (block) {
+			result++;
+		}
+	}
+	return result;
+}
+
+
+static int readYColumn(Map &map, s16 n)
+{
+	int result = 0;
+	for(s16 z=0; z<n; z++)
+	for(s16 x=0; x<n; x++)
+	for(s16 y=n-1; y>0; y--) {
+		v3s16 p(x,y,z);
+		MapBlock *block = map.getBlockNoCreateNoEx(p);
+		if (block) {
+			result++;
+		}
+	}
+	return result;
+}
+
+static int readNodes(Map &map, s16 n)
+{
+	int result = 0;
+	for(s16 z=0; z<n * MAP_BLOCKSIZE; z+=8)
+	for(s16 y=0; y<n * MAP_BLOCKSIZE; y+=4)
+	for(s16 x=0; x<n * MAP_BLOCKSIZE; x++) {
+		v3s16 p(x,y,z);
+		MapNode n = map.getNode(p);
+		if (n.getContent() != CONTENT_IGNORE)
+			result++;
+	}
+	return result;
+}
+
+
+#define BENCH1(_count) \
```

Comment:
No macro seems to be necessary for this, but that is unfortunately not obvious: See https://github.com/catchorg/Catch2/issues/1777#issuecomment-2992866943

Reply:
Just followed what the other benchmarks were doing.
Maybe commit this as is (assuming no other issues) and then we fix the macros wholesale across all benchmarks...?